lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -492,6 +526,14 @@ def fetch_ec2_metadata
       end
     end
 
+    def detect_cloudfunctions(attributes, common_labels)
+      return unless attributes.include?('gcf_region')
+      # Cloud Functions detected
+      @running_cloudfunctions = true
+      @gcf_region = fetch_gce_metadata('instance/attributes/gcf_region')
+      common_labels["#{CLOUDFUNCTIONS_SERVICE}/region"] = @gcf_region
```

Comment:
I think either way would be okay but it is indeed cleaner not to include this label if the service_name is not CLOUDFUNCTIONS_SERVICE eventually. So I am moving setting this label to the 'if match_data' block.